src/Donations/resources/hooks/useDonationDelete.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * @unreleased
+ */
+export function useDonationDelete() {
+    const [error, setError] = useState<Error | null>(null);
+    const [isResolving, setIsResolving] = useState(false);
+    const [hasResolved, setHasResolved] = useState(false);
+
+    async function deleteDonation(donationId: number, force: boolean = false) {
+      if (!donationId) return;
+
+      setIsResolving(true);
+      setHasResolved(false);
+      setError(null);
+
+      try {
+        await apiFetch({
+          path: `/givewp/v3/donations/${donationId}?force=${force}`,
+          method: 'DELETE',
+        });
```

Comment:
I discovered we can use [deleteEntityRecord](https://developer.wordpress.org/block-editor/reference-guides/data/data-core/#deleteentityrecord) here instead of fetch.

It will make a the DELETE request and we would include the force param.

There's a nifty little tutorial [here](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/data-basics/5-adding-a-delete-button.md) on how to use it but it's pretty similiar to our other entity record usages

Reply:
Since we were not using anything else from the useDonationDelete custom hook I replaced it with inline usages.
Resolved ca2d215780ca5ac2d7a0c291df22fa06ae0ea104

There was also an issue while deleting a donation where the pageTitle would show NAN so I pushed a quick fix - so that it does not display. 4fe90fb9066ddadfaf3bf069b02cbcb12f4dbb5e

Reply:
@JoshuaHungDinh awesome, glad that entity function worked!